parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestDataPageV1Checksums.java

```diff
@@ -0,0 +1,547 @@
+    file.delete();
+    Path path = new Path(file.toURI());
+
+    for (int i = 0; i < PAGE_SIZE; i++) {
+      colAPage1Bytes[i] = (byte) i;
+      colAPage2Bytes[i] = (byte) -i;
+      colBPage1Bytes[i] = (byte) (i + 100);
+      colBPage2Bytes[i] = (byte) (i - 100);
+    }
+
+    ParquetFileWriter writer =  new ParquetFileWriter(conf, schemaSimple, path,
+      ParquetWriter.DEFAULT_BLOCK_SIZE, ParquetWriter.MAX_PADDING_SIZE_DEFAULT);
+
+    writer.start();
+    writer.startBlock(numRecordsLargeFile);
+
+    CodecFactory codecFactory = new CodecFactory(conf, 1024 * 1024);
+    CodecFactory.BytesCompressor compressor = codecFactory.getCompressor(compression);
+
+    ColumnChunkPageWriteStore writeStore = new ColumnChunkPageWriteStore(
+      compressor, schemaSimple, new HeapByteBufferAllocator(),
+      Integer.MAX_VALUE, ParquetOutputFormat.getPageWriteChecksumEnabled(conf));
+
+    PageWriter pageWriter = writeStore.getPageWriter(colADesc);
+    pageWriter.writePage(BytesInput.from(colAPage1Bytes), numRecordsLargeFile / 2,
+      numRecordsLargeFile / 2, EMPTY_STATS_INT32, Encoding.RLE, Encoding.RLE, Encoding.PLAIN);
+    pageWriter.writePage(BytesInput.from(colAPage2Bytes), numRecordsLargeFile / 2,
+      numRecordsLargeFile / 2, EMPTY_STATS_INT32, Encoding.RLE, Encoding.RLE, Encoding.PLAIN);
+
+    pageWriter = writeStore.getPageWriter(colBDesc);
+    pageWriter.writePage(BytesInput.from(colBPage1Bytes), numRecordsLargeFile / 2,
+      numRecordsLargeFile / 2, EMPTY_STATS_INT32, Encoding.RLE, Encoding.RLE, Encoding.PLAIN);
+    pageWriter.writePage(BytesInput.from(colBPage2Bytes), numRecordsLargeFile / 2,
+      numRecordsLargeFile / 2, EMPTY_STATS_INT32, Encoding.RLE, Encoding.RLE, Encoding.PLAIN);
+
+    writeStore.flushToFileWriter(writer);
+
+    writer.endBlock();
+    writer.end(new HashMap<>());
+
+    codecFactory.release();
+
+    return path;
+  }
+
+  // Sample data, nested schema with nulls
+
+  private static final MessageType schemaNestedWithNulls = MessageTypeParser.parseMessageType(
+    "message m {" +
+      "  optional group c {" +
+      "    required int64 id;" +
+      "    required group d {" +
+      "      repeated int32 val;" +
+      "    }" +
+      "  }" +
+      "}");
+  private static final ColumnDescriptor colCIdDesc = schemaNestedWithNulls.getColumns().get(0);
+  private static final ColumnDescriptor colDValDesc = schemaNestedWithNulls.getColumns().get(1);
+
+  private static final double nullRatio = 0.3;
+  private static final int numRecordsNestedWithNullsFile = 2000;
+
+  private Path writeNestedWithNullsSampleParquetFile(Configuration conf,
+                                                     boolean dictionaryEncoding,
+                                                     CompressionCodecName compression)
+    throws IOException {
+    File file = tempFolder.newFile();
+    file.delete();
+    Path path = new Path(file.toURI());
+
+    ParquetWriter<Group> writer = ExampleParquetWriter.builder(path)
```

Comment:
Can we wrap a try-with-resource around this

Reply:
Again good catch, thanks